plugin/providers/runscope/data_source_runscope_integrations_test.go

```diff
@@ -0,0 +1,149 @@
+	"os"
+	"testing"
+)
+
+func TestAccDataSourceRunscopeIntegrations_Basic(t *testing.T) {
+
+	teamId := os.Getenv("RUNSCOPE_TEAM_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccDataSourceRunscopeIntegrationsConfig, teamId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceRunscopeIntegrations("data.runscope_integrations.by_type"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceRunscopeIntegrations(dataSource string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		r := s.RootModule().Resources[dataSource]
+		a := r.Primary.Attributes
+
+		if a["ids.#"] != "2" {
+			return fmt.Errorf("Expected to get integrations returned from runscope data resource")
+		}
+
+		if a["ids.0"] == "" {
+			return fmt.Errorf("Expected to get an integration ID from runscope data resource")
+		}
+
+		return nil
+	}
+}
+
+const testAccDataSourceRunscopeIntegrationsConfig = `
+data "runscope_integrations" "by_type" {
+	team_uuid = "%s"
+	filter = {
+		name = "type"
+		values = ["slack"]
+	}
+}
+`
+
+func TestAccDataSourceRunscopeIntegrations_usage(t *testing.T) {
+
+	teamId := os.Getenv("RUNSCOPE_TEAM_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccDataSourceRunscopeIntegrationsUsageConfig, teamId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEnvironmentIntegrations("runscope_environment.environment_with_integrations", true),
+					testAccCheckEnvironmentIntegrations("runscope_environment.environment_no_integrations", false),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckEnvironmentIntegrations(environment string, expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[environment]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", environment)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Record ID is set")
+		}
+
+		client := testAccProvider.Meta().(*runscope.Client)
+
+		var foundRecord *runscope.Environment
+		var err error
+
+		environment := new(runscope.Environment)
+		environment.ID = rs.Primary.ID
+		bucketId := rs.Primary.Attributes["bucket_id"]
+		foundRecord, err = client.ReadSharedEnvironment(environment,
+			&runscope.Bucket{Key: bucketId})
+
+		if err != nil {
+			return err
+		}
+
+		if len(foundRecord.Integrations) == 0 && expected {
+			return fmt.Errorf("Expected environment to have integrations")
+		} else if len(foundRecord.Integrations) != 0 && !expected {
+			return fmt.Errorf("Expected environment not to have integrations, but had %d", len(foundRecord.Integrations))
+		}
+
+		return nil
+	}
+}
+
+const testAccDataSourceRunscopeIntegrationsUsageConfig = `
+data "runscope_integrations" "slack" {
```

Comment:
@philwhiteuk assuming you have exactly 2 slack integrations configured in the team that this test uses?